spec/services/authn_context_resolver_spec.rb

```diff
@@ -427,4 +427,281 @@
       end
     end
   end
+
+  context 'when resolving semantic acr_values' do
+    before do
+      allow(IdentityConfig.store).
+        to receive(:feature_valid_authn_contexts_semantic_enabled).
+        and_return(true)
+      allow_any_instance_of(ServiceProvider).
+        to receive(:semantic_authn_contexts_allowed?).
+        and_return(true)
+
+      stub_const(
+        'Saml::Idp::Constants::VALID_AUTHN_CONTEXTS',
+        IdentityConfig.store.valid_authn_contexts_semantic,
+      )
+    end
+
+    context 'when no semantic ACR present' do
+      let(:user) { build(:user) }
+      it 'does not resolve legacy ACRs to semantic ACRs' do
+        acr_values = [
+          Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
+        ]
+
+        resolver = AuthnContextResolver.new(
```

Comment:
can we pull out the resolver as a subject and the acr_values into let statements so we don't need to duplicate the setup in all these specs? (similar to how you have it structured below)

Reply:
you would have to pull these all into context blocks, ie

```
context 'legacy ACR values are user' do 
  let(:acr_values) do
    [
      Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF,
    ]

  it 'resolves to legacy ACRs' do
   ..
  end
end
```

Reply:
I had started to do that, but decided to not introduce more noise the PR than already exists. Thoughts?

Reply:
i think it could have just been done in this context block, but it's fine how it is, we can clean up in a followup